src/review/review.controller.ts

```diff
@@ -0,0 +1,79 @@
+import { Controller, Get, Post, Body, Param, UseGuards } from '@nestjs/common';
+import { ReviewService } from './review.service';
+import { Public } from 'src/common/decorators/publicRoute.decorator';
+import { IdCheck } from 'src/common/decorators/idcheck.decorator';
+import {
+  ApiBearerAuth,
+  ApiOperation,
+  ApiQuery,
+  ApiTags,
+} from '@nestjs/swagger';
+import { Ok, ZaLaResponse } from 'src/common/helpers/response';
+import { ApiRatingDto } from 'src/review/dto/create-api-review.dto';
+import { AuthenticationGuard } from 'src/common/guards/authentication.guard';
+import { Review } from './../entities/review.entity';
+import { Paginate, Paginated, PaginateQuery } from 'nestjs-paginate';
+
+@ApiTags('Review')
+@UseGuards(AuthenticationGuard)
+@ApiBearerAuth('access-token')
+@Controller('review')
+export class ReviewController {
+  constructor(private readonly reviewService: ReviewService) {}
+
+  @IdCheck('apiId', 'profileId')
+  @Post(':apiId/:profileId')
+  @ApiOperation({ summary: 'Review an api' })
+  async addApiRating(
+    @Param('profileId') profileId: string,
+    @Param('apiId') apiId: string,
```

Comment:
This is fine, But you can get the profileId from global request object, minimizes amount of code that way.

Reply:
Yeah. I understand. The thing is just that i won't be able to test since I am removing the authentication before i test and the profileId can't be obtained that way. (I am still having issue with authentication; that's the reason for me removing the authentication before i test)